apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs

```diff
@@ -491,7 +491,18 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
             if (SendJsonPayload)
             {
                 var jsonStruct = new Struct();
+                if (EnableJsonLayout)
+                {
+                    foreach (var jToken in JObject.Parse(RenderLogEvent(Layout, loggingEvent)))
+                    {
+                        jsonStruct.Fields.Add(jToken.Key, ProtoConverter.Convert(jToken.Value));
+                    }
+                    logEntry.JsonPayload = jsonStruct;
+                    return FillLogEntryCommonData(loggingEvent, logEntry);
```

Comment:
I think it would be clearer just to add an "else" for the normal case here, and let `FillLogEntryCommonData` occur at the end of the method. The current structure confused me for a bit.

Reply:
Hmm... If someone wants to trace the `EnableJsonLayout` case then he can finish reading the method at this place instead of looking further. In this case, the 'else' statement is redundant. So... maybe it's a little bit better then?

If I understood you correctly you also prefer to check this condition near the properties and service context structure filling?

My goal was to avoid double-checking (or even triple) the same scenario. But maybe I got you wrong and you want to nest everything in else statement?

Reply:
>  In this case, the 'else' statement is redundant. 

Yes, but it involves the duplication of the `FillLogEntryCommonData` return statement. So in order to check the differences between EnableJsonLayout being set or not, you have to look quite hard IMO. What I'm suggesting is removing this `return` statement, and putting everything that happens when EnableJsonLayout is *disabled* in an `else` block. That keeps the two sets of either/or statements at the same level of nesting, making it clearer. It's easy to miss the return statement when skimming, and assume that the EnableJsonLayout code is *in addition* rather than *instead of* the rest.